File: src/main/java/com/example/trainingsystem/models/Trainee_Course_Management.java
package com.example.trainingsystem.models;

import com.example.trainingsystem.models.embedded.Content_TCManagement;
import com.example.trainingsystem.models.embedded.FinalExam_TCManagement;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Document(collection = "Trainee_Course_Management")
public class Trainee_Course_Management {
    @Id
    @JsonProperty("_id")
    private String _id;

    @JsonProperty("TraineeName")
    private String TraineeName;

    @JsonProperty("CourseName")
    private String CourseName;

    @JsonProperty("Contents")
    private ArrayList<Content_TCManagement> Contents;
    //Add ContentName
    //Add Documentation
    //Add Duration
    //Add Exercises (Array of Obj: ExerciseDescription + Type + Submission + Mark + Weight)

    @JsonProperty("FinalExam")
    private FinalExam_TCManagement FinalExam;
    //Add Type + Submission + Mark1st + Mark2nd + Weight

    @JsonProperty("TotalScore")
    private double TotalScore;

    @JsonProperty("Status")
    private String Status;

    @JsonProperty("Level")
    private int Level;

    @JsonProperty("EnrollmentDate")
    private LocalDateTime EnrollmentDate;

    @JsonProperty("CompletedDate")
    private LocalDateTime CompletedDate;

    @JsonProperty("Rate")
    private int Rate;
}

File: TrainingSystem.NetAPI/MongoDB_CRUD/Models/TraineeCourseManagement.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;
using System;


namespace MongoDB_CRUD.Models
{
    public class TraineeCourseManagement
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("TraineeName")]
        public string TraineeName { get; set; }
        [BsonElement("CourseName")]
        public string CourseName { get; set; }
        [BsonElement("Contents")]
        public List<Content> Contents { get; set; }
        [BsonElement("FinalExam")]
        public FinalExam FinalExam { get; set; }
        [BsonElement("TotalScore")]
        public double TotalScore { get; set; }
        [BsonElement("Status")]
        public string Status { get; set; }
        [BsonElement("Level")]
        public int Level { get; set; }
        [BsonElement("EnrollmentDate")]
        public DateTime EnrollmentDate { get; set; }
        [BsonElement("CompletedDate")]
        public DateTime CompletedDate { get; set; }
        [BsonElement("Rate")]
        public int Rate { get; set; }
        [BsonElement("_class")]
        public string _class { get; set; }


    }
}
